project/src/test/java/com/google/sps/data/PlacesFetcherTest.java

```diff
@@ -209,4 +209,18 @@ public void fetch_PlaceDetailsQueryFails_throwsFetcherException() throws Excepti
     assertTrue(thrown.getCause() instanceof IOException);
     assertTrue(thrown.getMessage().contains("place details"));
   }
+
+  @Test
+  public void fetch_resultsOnlyAfterRadiusExtension_returnsListOfPlaces() throws Exception {
```

Comment:
Add a test that retrieves nothing after max number of extensions was reached to handle the edge case.

Reply:
The fetch_zeroSearchResults_returnsEmptyList test checks this corner case. In this test the getPlacesSearchResults always returns nothing so this test ends only when max number of extensions were reached and recieves nothing.

Reply:
I see, in that case please add a verification in the test that the mock was called MAX_NUM_OF_RADIUS_EXTENSIONS. You can use mockito's verify method, something like:
Mockito.verify(spiedFetcher, times(MAX_NUM_OF_RADIUS_EXTENSIONS)).getPlacesSearchResults()

see https://javapointers.com/java/unit-test/use-verify-in-mockito/ for more info.